File: 0067-add-binary/0067-add-binary.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        string res="";
        int c=0;
        
        for(int i=a.size()-1,j=b.size()-1;i>=0||j>=0;i--,j--){
            int sum=(i>=0?a[i]-'0':0)+(j>=0?b[j]-'0':0)+c;
            if(sum==0){
                res+="0";
            }
            else if(sum==1){
                res+="1";
                c=0;
            }
            else if(sum==2){
                res+="0";
                c=1;
            }
            else{
                res+="1";
                c=1;
            }
            
        }
        if(c)
            res+="1";
        
        reverse(res.begin(),res.end());
        return res;
    }
};

File: 0067-add-binary/0067-add-binary.java
class Solution {
    public String addBinary(String a, String b) {
        
        String res="";
        int i=a.length()-1,j=b.length()-1,carry=0;
        
        while(i>=0 || j>=0 || carry==1){
            int curr=carry;
            carry=0;
            
            if(i>=0 && a.charAt(i)=='1'){
                curr+=1;
            }
            if(j>=0 && b.charAt(j)=='1'){
                curr+=1;
            }
            
            if(curr==0){
                res="0"+res;
            }
            else if(curr==1){
                res="1"+res;
            }
            else if(curr==2){
                res="0"+res;
                carry=1;
            }
            else{
                res="1"+res;
                carry=1;
            }
            
            --i;
            --j;
        }
        
        return res;
    }
}